util/fipstools/acvp/acvptool/subprocess/hash.go

```diff
@@ -118,32 +120,56 @@ func (h *hashPrimitive) Process(vectorSet []byte, m Transactable) (any, error) {
 					panic(h.algo + " hash operation failed: " + err.Error())
 				}
 
-				response.Tests = append(response.Tests, hashTestResponse{
+				testResponse := hashTestResponse{
 					ID:        test.ID,
 					DigestHex: hex.EncodeToString(result[0]),
-				})
-
-			case "MCT":
-				// MCT tests for conventional digest functions expect message
-				// and digest output lengths to be equivalent, however SHAKE
-				// does not have a predefined output length.
-				if len(msg) != h.size && !strings.HasPrefix(h.algo, "SHAKE") {
-					return nil, fmt.Errorf("MCT test case %d/%d contains message of length %d but the digest length is %d", group.ID, test.ID, len(msg), h.size)
+				}
+				if test.OutputLength != nil {
+					testResponse.OutLen = *test.OutputLength
 				}
 
-				testResponse := hashTestResponse{ID: test.ID}
+				response.Tests = append(response.Tests, testResponse)
 
-				digest := msg
-				for i := 0; i < 100; i++ {
-					result, err := m.Transact(h.algo+"/MCT", 1, digest)
-					if err != nil {
-						panic(h.algo + " hash operation failed: " + err.Error())
+			case "MCT":
+				testResponse := hashTestResponse{ID: test.ID}
+				if !strings.HasPrefix(h.algo, "SHAKE") {
+					if len(msg) != h.size {
+						return nil, fmt.Errorf("MCT test case %d/%d contains message of length %d but the digest length is %d", group.ID, test.ID, len(msg), h.size)
+					}
+					digest := msg
+					for i := 0; i < 100; i++ {
+						result, err := m.Transact(h.algo+"/MCT", 1, digest)
+						if err != nil {
+							panic(h.algo + " hash operation failed: " + err.Error())
+						}
+
+						digest = result[0]
+						testResponse.MCTResults = append(testResponse.MCTResults, hashMCTResult{hex.EncodeToString(digest), 0})
 					}
 
-					digest = result[0]
-					testResponse.MCTResults = append(testResponse.MCTResults, hashMCTResult{hex.EncodeToString(digest)})
-				}
+				} else {
+					maxOutLenBytes := *group.MaxOutLen / 8
+					minOutLenBytes := *group.MinOutLen / 8
+
+					digest := msg
+					outlen := maxOutLenBytes
 
+					maxOutLenByteArr := uint32le(uint32(maxOutLenBytes))
```

Comment:
Why the conversion explicitly to little endian?

Reply:
1. because its how its done elsewhere in the acvp code
2. the modulewrapper relies on it being a byte array in little endian format, so its better to be explicit here in my opinion